src/mon/OSDMonitor.cc

```diff
@@ -8417,6 +8458,204 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
 						get_last_committed() + 1));
       return true;
     }
+  } else if (prefix == "osd add-noout-osds" ||
+             prefix == "osd add-nodown-osds") {
+
+    bool noout = prefix == "osd add-noout-osds";
+    bool any = false;
+
+    vector<string> idvec;
+    cmd_getval(g_ceph_context, cmdmap, "ids", idvec);
+    for (unsigned j = 0; j < idvec.size(); j++) {
+      long osd_start = -1;
+      long osd_end = -1;
```

Comment:
nit: usually use "end" in exclusive sense, like [begin, end).. for inclusive maybe [first,last].